addons/partner_autocomplete/tests/test_res_company.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import common
+
+
+class TestResCompany(common.TransactionCase):
+
+    def test_extract_company_domain(self):
+        company_1 = self.env['res.company'].create({'name': "Test Company 1"})
+
+        company_1.website = 'http://www.info.proximus.be/faq/test'
+        self.assertEqual(company_1._get_company_domain(), "proximus.be")
+
+        company_1.email = 'info@waterlink.be'
+        self.assertEqual(company_1._get_company_domain(), "waterlink.be")
+
+        company_1.website = False
+        company_1.email = False
+        self.assertEqual(company_1._get_company_domain(), False)
+
+        company_1.email = "at@"
+        self.assertEqual(company_1._get_company_domain(), False)
+
+        company_1.website = "http://superFalsyWebsiteName"
+        self.assertEqual(company_1._get_company_domain(), False)
+
+        company_1.website = "http://www.superwebsite.com"
+        self.assertEqual(company_1._get_company_domain(), 'superwebsite.com')
+
+        company_1.website = "http://superwebsite.com"
+        self.assertEqual(company_1._get_company_domain(), 'superwebsite.com')
+
+        company_1.website = "http://localhost:8069/%7Eguido/Python.html"
+        self.assertEqual(company_1._get_company_domain(), False)
+
+        company_1.website = "http://runbot.odoo.com"
+        self.assertEqual(company_1._get_company_domain(), 'odoo.com')
```

Comment:
I still think this is a good example of the problem, you can't assume that `coca.odoo.com` and `pepsi.odoo.com` belongs to the same company (all your coke belong to us!)

Reply:
@mart-e Feel free to make a feedback to feedback@mail.odoo.com . I bet it will be managed with as much kindness as necessary.